### ISSUE TITLE
v2: NettyClientTest not unit tests

### ISSUE THREAD

**Issue by davidmoten** [Azure/autorest-clientruntime-for-java#403]
`NettyClientTest ` appears to access external resources (`https://httpbin.org/get`) so may not be a unit test. 

The tests hung for me because I am behind a proxy server and of course would not work at all if I was not connected to the internet.

I suggest that some dummy web server gets started up locally by the unit tests. 

Just thought I'd add this to your backlog!

**Comment by RikkiGibson** [comment]
Yeah, hitting httpbin.org for all our basic testing just happened to be expedient for us up to this point. A local web server seems like the most foolproof thing in the long term.

Would it be reasonable to have a script spin up a local [httpbin](https://github.com/kennethreitz/httpbin) instance? Only downside is people may need to install Python.

**Comment by RikkiGibson** [comment]
Otherwise, some Java-based test server that implements the parts of httpbin that we actually use might not be very hard to throw together. Would probably be happier with this as a test dependency for a project that we intend to distribute.

**Comment by davidmoten** [comment]
Just tried WireMock which starts up a jetty server  on a free port and shuts down using JUnit rule. Not invasive, pretty good at first look:

```
public class NettyClientTests {

    @Rule
    public WireMockRule server = new WireMockRule(0);

    @Test
    public void testRequestBeforeShutdownSucceeds() throws Exception {
        server.stubFor(WireMock.get("/get").willReturn(WireMock.aResponse().withBody("hi there")));

        final HttpClientFactory factory = new NettyClient.Factory();
        HttpClient client = factory.create(null);
        HttpRequest request = new HttpRequest("", HttpMethod.GET, url(server, "/get"), null);

        client.sendRequestAsync(request).blockingGet();
        factory.close();
    }
```
    private static URL url(WireMockRule server, String path) {
        try {
            return new URL("http://localhost:" + server.port() + path);
        } catch (MalformedURLException e) {
            throw new RuntimeException(e);
        }
    }

**Comment by RikkiGibson** [comment]
Looks good. I might take a stab at modifying the RestProxy tests to use this.

**Comment by RikkiGibson** [comment]
Addressed in #406-- more changes to make tests depend less on external services might be wise
